src/hotspot/share/gc/g1/g1ConcurrentRefineThreadsNeeded.cpp

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#include "precompiled.hpp"
+#include "gc/g1/g1Analytics.hpp"
+#include "gc/g1/g1ConcurrentRefineThreadsNeeded.hpp"
+#include "gc/g1/heapRegion.hpp"
+#include "gc/g1/g1Policy.hpp"
+#include "utilities/globalDefinitions.hpp"
+#include <math.h>
+
+G1ConcurrentRefineThreadsNeeded::G1ConcurrentRefineThreadsNeeded(G1Policy* policy,
+                                                                 double update_period_ms) :
+  _policy(policy),
+  _update_period_ms(update_period_ms),
+  _predicted_time_until_next_gc_ms(0.0),
+  _predicted_cards_at_next_gc(0),
+  _threads_needed(0)
+{}
+
+// Estimate how many concurrent refinement threads we need to run to achieve
+// the target number of card by the time the next GC happens.  There are
+// several additional desirements we'd like to achieve while meeting that
```

Comment:
According to https://en.wiktionary.org/wiki/desirement this is defined as

  Something that is [desired](https://en.wiktionary.org/wiki/desired), but not absolutely required.

(This word was new to me).

I think that "desirements" already includes the phrase "we'd like to achieve". I.e. something that is desired, but is not absolutely required is already something "we'd like to achieve".

So I would like to suggest to either remove the phrase "we'd like to achieve" or reformulate the sentence as "... several additional/secondary goals we would like to achieve while meeting that (main) goal".

Reply:
Done.  Using "secondary goals".